asteroid/masknn/attention.py

```diff
@@ -151,11 +168,11 @@ def __init__(
                     ]
                 )
             )
-        net_out_conv = nn.Conv2d(self.in_chan, n_src * self.in_chan, 1)
+        net_out_conv = nn.Conv2d(self.mha_in_dim, n_src * self.mha_in_dim, 1)
         self.first_out = nn.Sequential(nn.PReLU(), net_out_conv)
         # Gating and masking in 2D space (after fold)
-        self.net_out = nn.Sequential(nn.Conv1d(self.in_chan, self.in_chan, 1), nn.Tanh())
-        self.net_gate = nn.Sequential(nn.Conv1d(self.in_chan, self.in_chan, 1), nn.Sigmoid())
+        self.net_out = nn.Sequential(nn.Conv1d(self.mha_in_dim, self.mha_in_dim, 1), nn.Tanh())
+        self.net_gate = nn.Sequential(nn.Conv1d(self.mha_in_dim, self.mha_in_dim, 1), nn.Sigmoid())
```

Comment:
The mask should be computed with `in_chan`, not with `mha_in_dim`. So actually, we don't need an output linear layer, but all these layers will `use `in_chan`, from `net_out_conv`.